pkg/controller/controller_instance_test.go

```diff
@@ -821,6 +822,119 @@ func TestReconcileServiceInstanceDelete(t *testing.T) {
+	binding := getTestServiceInstanceCredential()
+	sharedInformers.ServiceInstanceCredentials().Informer().GetStore().Add(binding)
+
+	instance := getTestServiceInstance()
+	instance.ObjectMeta.DeletionTimestamp = &metav1.Time{}
+	instance.ObjectMeta.Finalizers = []string{v1alpha1.FinalizerServiceCatalog}
+	// we only invoke the broker client to deprovision if we have a checksum set
+	// as that implies a previous success.
+	checksum := checksum.ServiceInstanceSpecChecksum(instance.Spec)
+	instance.Status.Checksum = &checksum
+
+	fakeCatalogClient.AddReactor("get", "instances", func(action clientgotesting.Action) (bool, runtime.Object, error) {
+		return true, instance, nil
+	})
+
+	err := testController.reconcileServiceInstance(instance)
+	if err != nil {
+		t.Fatalf("This should not fail")
+	}
+
+	brokerActions := fakeBrokerClient.Actions()
+	assertNumberOfServiceBrokerActions(t, brokerActions, 0)
+
+	// Verify no core kube actions occurred
+	kubeActions := fakeKubeClient.Actions()
+	assertNumberOfActions(t, kubeActions, 0)
+
+	actions := fakeCatalogClient.Actions()
+	// The action should be Update to Ready = false
+	assertNumberOfActions(t, actions, 1)
+
+	updatedServiceInstance := assertUpdateStatus(t, actions[0], instance)
+	assertServiceInstanceReadyFalse(t, updatedServiceInstance)
+
+	events := getRecordedEvents(testController)
+	assertNumEvents(t, events, 1)
+
+	expectedEvent := api.EventTypeWarning + " " + "DeprovisionCallFailed Delete instance test-ns/test-instance blocked by existing binding test-ns/test-instance.  All bindings must be removed first."
+	if e, a := expectedEvent, events[0]; e != a {
+		t.Fatalf("Received unexpected event: %v", a)
+	}
+
+	// delete binding
+	sharedInformers.ServiceInstanceCredentials().Informer().GetStore().Delete(binding)
+
+	fakeKubeClient.ClearActions()
+	fakeCatalogClient.ClearActions()
+
+	// Binding was removed, verify the next reconcilation removes
+	// the instance
+
+	err = testController.reconcileServiceInstance(instance)
+	if err != nil {
+		t.Fatalf("This should not fail")
+	}
+
+	brokerActions = fakeBrokerClient.Actions()
+	assertNumberOfServiceBrokerActions(t, brokerActions, 1)
+	assertDeprovision(t, brokerActions[0], &osb.DeprovisionRequest{
+		AcceptsIncomplete: true,
+		InstanceID:        instanceGUID,
+		ServiceID:         serviceClassGUID,
+		PlanID:            planGUID,
+	})
+
+	// Verify no core kube actions occurred
+	kubeActions = fakeKubeClient.Actions()
+	assertNumberOfActions(t, kubeActions, 0)
+
+	actions = fakeCatalogClient.Actions()
+
+	// The three actions should be:
+	// 0. Updating the ready condition
+	// 1. Get against the instance
+	// 2. Removing the finalizer
+	assertNumberOfActions(t, actions, 3)
+
+	updatedServiceInstance = assertUpdateStatus(t, actions[0], instance)
+	assertServiceInstanceReadyFalse(t, updatedServiceInstance)
+
+	assertGet(t, actions[1], instance)
+
+	// JAY: BOTH OF THESE ARE FAILING AFTER LATEST MERGE
```

Comment:
@pmorie  any thoughts about why I'm seeing these failures documented here on lines 921 - 927?  This was working prior to a merge I did Friday afternoon.   Walking through the reconcile() it looks to me like its doing the right thing including deleting the finalizers.